cmd/skopeo/sync.go

```diff
@@ -758,7 +774,18 @@ func (opts *syncOptions) run(args []string, stdout io.Writer) (retErr error) {
 			} else {
 				logrus.WithFields(fromToFields).Infof("Copying image ref %d/%d", counter+1, len(srcRepo.ImageRefs))
 				if err = retry.IfNecessary(ctx, func() error {
-					_, err = copy.Image(ctx, policyContext, destRef, ref, &options)
+					manifestBytes, err := copy.Image(ctx, policyContext, destRef, ref, &options)
+					// Ensure that we log the manifest digest to a file only if the copy operation was successful
+					if opts.digestFile != "" && err == nil {
+						manifestDigest, err := manifest.Digest(manifestBytes)
+						if err != nil {
+							return err
+						}
+						outputStr := fmt.Sprintf("%s %s", manifestDigest.String(), transports.ImageName(destRef))
+						if _, err = digestFile.WriteString(outputStr + "\n"); err != nil {
+							return fmt.Errorf("Failed to write digest to file %q: %w", opts.digestFile, err)
+						}
+					}
```

Comment:
Actually, all of this would be better outside of the `retry.IfNecessary` block; e.g. if the write to `digestFile` failed, we should terminate immediately, and definitely not restart the image copy.

Reply:
Done.